Cabal/Distribution/PackageDescription/Check.hs

```diff
@@ -1186,6 +1186,23 @@ checkCabalVersion pkg =
            [ display (Dependency name (eliminateMajorBoundSyntax versionRange))
            | Dependency name versionRange <- depsUsingMajorBoundSyntax ]
 
+  , checkVersion [2,1] (any (not . null)
+                        (concatMap buildInfoField
+                         [ asmSources
+                         , cmmSources
+                         , extraBundledLibs
+                         , extraLibFlavours ])) $
+      PackageDistInexcusable $
+           "The use of 'asm-sources', 'cmm-sources', 'extra-bundled-libraries' "
+        ++ " and 'virtual-modules' requires the package "
```

Comment:
Oops, looks like you need to `s/virtual-modules/extra-library-flavours/` here. Feel free to commit directly to `master`.

Reply:
wow. How the hell did that happen? Ok... I'll just edit it on GH.